ui: database pagination

vladlos opened the pull request:
Added pagination for database tables, removed database info stats, they will not work with pagination correctly.

Resolves: #43026

Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/44009)
<!-- Reviewable:end -->

vladlos committed:
ui: database pagination

Added pagination for database tables, removed database info stats, they will not work with pagination correctly.

Resolves: #43026

Release note(ui): none

dhartunian reviewed (changes requested):
@elkmaster this change breaks the databases/grants page. Can you take a look there? The two pages share the `DatabaseSummaryBase` component.

dhartunian commented:
Also @elkmaster feel free to modify the design of `DatabaseSummaryBase` if that makes things easier.

dhartunian commented on pkg/ui/src/views/databases/containers/databaseGrants/index.tsx:
```diff
@@ -49,21 +75,28 @@ class DatabaseSummaryGrants extends DatabaseSummaryBase {
-                      },
-                    ]} />
+                  <React.Fragment>
+                    <DatabaseGrantsSortedTable
```
@elkmaster Can we combine the `PaginationComponent` and `DatabaseGrantsSortedTable` into a `PaginatedDatabaseGrantsSortedTable`? See if it's straightforward to do. If it's complicated I'm ok with leaving as-is.

Extra challenge: can we create a generic `PaginatedTable` that's parametrized with the table type?

elkmaster replied:
If we create `PaginatedDatabaseGrantsSortedTable` it will look very similar to existing file `databaseGrants/index.tsx`. I would left this as-is, as we still will be restructuring table views, after we move\create table component in upcoming design system. Creating generic PaginatedTable that's parameterized, also make more sense doing after, and task big enough to have at least it own ticket.

dhartunian commented:
@elkmaster can you rebase on master? I think that will help tests pass.

vladlos committed:
ui: database grants

added pagination to grants tables

Release note (ui): none

dhartunian reviewed (approved):
@elkmaster I'm merging this for now since we need to move on to the database redesign anyway and this is setting the stage for that.

I do want to point out that in the future, adding new functionality should not take any existing pieces away. I understand how the database stats box could have been tricky to integrate with pagination in this instance but in that case we need to refactor the code to make such a change possible without disturbing existing functionality.

In this case, because we're redesigning the pages in the next few weeks it's not a big problem.

dhartunian commented:
bors r+
